htdocs/core/class/html.formticket.class.php

```diff
@@ -264,23 +266,26 @@ public function showForm($withdolfichehead = 0, $mode = 'edit', $public = 0, Con
                             jQuery("#company_name").val("");
                             jQuery("#contact_phone").val("");
 
-                            jQuery.getJSON(
+                            jQuery.post(
                                 "'.dol_escape_js(dol_buildpath('/public/ticket/ajax/ajax.php', 1)).'",
 								{
 									action: "getContacts",
-									email: jQuery("#email").val()
+									email: jQuery("#email").val(),
+									token: "'.currentToken().'"
 								},
 								function(response) {
+									var response = jQuery.parseJSON(response);
 									if (response.error) {
                                         jQuery("#contact_search_result").html("<span class=\"error\">"+response.error+"</span>");
 									} else {
                                         var contact_list = response.contacts;
-										if (contact_list.length == 1) {
+
+										if (typeof(contact_list) !== "undefined" && contact_list.length) {
```

Comment:
What happen if the response return 2 record ?

Reply:
@eldy if 2 records with the same email address it displays the first, it's not good I presume? :-)

Reply:
@eldy if nothing is displayed when it finds more than 1, there will be the possibility of creating a new contact at the risk of creating one at each request!

Reply:
@eldy I just tested with "contact_list.length == 1" and indeed with each ticket request the user creates a new third party with the same name!

Reply:
@eldy to have this type of functionality it would be preferable to have a customer account access and a customer account request!

Reply:
> @eldy to have this type of functionality it would be preferable to have a customer account access and a customer account request!

Sure. Autofill data based on a user input with outside of a login session will always means "no good duplicate management" and also "no security". It means you can guess and retreive data by entering random entry. 

On other pages, you must first enter   "A Public Tracking ID" and an "Email" and email must match the email of the ticket of the entered public tracking id to be into a valid session and see data. It should be the same for the call of this page. If we want to retreive a contact, search should not be done with email only but with couple email + a tracking id of a ticket previously entered.
This is why this feature `"if ($with_contact) {" ` is disabled by default.